src/components/molecules/event/eventBody/EventBody.tsx

```diff
@@ -112,21 +110,47 @@ export const EditEvent: React.FC<{ event: Event; setEdit: () => void }> = ({
       return;
     }
     try {
-      /* Check both file and payload as we the set error does not trigger if the 
-         file is set but the payload is not, and we cant send an empty payload*/
       if (file) {
-        const data = new FormData();
-        data.append('image', file, file.name);
-        await uploadEventPicture(event.eid, data);
+        await uploadEventPicture(event.eid, file);
       }
+    } catch (error) {
+      addToast({
+        title: 'Feil ved opplasting av bilde',
+        status: 'error',
+      });
+      return;
+    }
+    try {
+      /* Check both file and payload as we the set error does not trigger if the 
+         file is set but the payload is not, and we cant send an empty payload*/
       if (Object.keys(updatePayload).length) {
         await updateEvent(event.eid, updatePayload);
       }
-      // reload page
-      history.go(0);
     } catch (error) {
-      console.log(error);
+      let detail = await error.getText();
+      if (error.statusCode === 400) {
+        addToast({
+          title: 'Ikke godkjent oppdatering',
+          status: 'error',
+          description: `${detail}`,
+        });
+        return;
+      }
+      addToast({
```

Comment:
Do we need a return after this? Won't it show the success toast right after this if not?

Reply:
yeah didn't see that :+1: